### PR TITLE
Feat/docs/getting started

### PR DESCRIPTION
**Pull request by laSinteZ**
Signed-off-by: Vyacheslav Bikbaev la.sintez@gmail.com

### Description of the Change
Implemented documentation about "getting started"

### Benefits
Now we have a simple guide about how to start Iroha and send transactions

### Possible Drawbacks 
None

### CODE REVIEW

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  libc6-dbg golang \
+  git ssh tar gzip ca-certificates \
+  wget curl file unzip \
+  python
+
+.. note::  If you are willing to actively develop Iroha, please consider installing the `latest release <https://cmake.org/download/>`_ of CMake.
+
+macOS
+^^^^^
+If you are a happy macOS user and are willing to play with Iroha without building it, you can install our official Homebrew package and go straight to `Playing around with settings` section
```
[the remark this answers is not in the document]
Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: The only correction we need now is to remove version tag from ruby code. Can you help with that?

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+Getting Iroha build
+-------------------
+
+Cloning the repository
+^^^^^^^^^^^^^^^^^^^^^^
+Clone the `Iroha repository <https://github.com/hyperledger/iroha>`_ to the directory of your choice.
+
+.. code-block:: shell
+
+  git clone -b develop https://github.com/hyperledger/iroha
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: you can put a hint telling about --depth parameter

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+  cd iroha
+
+.. hint:: If you have installed the prerequisites with Docker, you don't need to clone Iroha again
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: I don't understand what you mean here

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+
+Building Iroha
+^^^^^^^^^^^^^^
+To build Iroha, use those commands
+
+.. code-block:: shell
+
+  cmake -H. -Bbuild;
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: also:
```
mkdir build; cd build; cmake ..; make -j$(nproc)
```

Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: this is maybe a bit easier to remember, but is just a reference

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+  cmake --build build -- -j$(nproc)
+
+.. note::  On macOS ``$(nproc)`` variable does not work. Check number of logical cores with ``sysctl -n hw.ncpu`` and put it explicitly in the command above, e.g. ``cmake --build build -- -j4``
+
+CMake parameters
+^^^^^^^^^^^^^^^^
+
+We use CMake to build platform-dependent build files. It has numerous of the flags for configuring the final build. Note that beside the listed params cmake's vars can be useful as well. Also as long as this page can be deprecated (or just not complete) you can browse custom flags via ``cmake -L``, ``cmake-gui``, or ``ccmake``.
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: https://www.englishforums.com/English/NumerousOfThe/qqmll/post.htm

has numerous flags

Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: besides the listed

https://dictionary.cambridge.org/grammar/british-grammar/beside-or-besides

Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: params — parameters
vars — variables

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+
+.. hint::  You can specify parameters at the cmake configuring stage (e.g cmake -DTESTING=ON).
+
+Main params
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: params — parameters

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+"""""""""""
+
++--------------+-----------------+---------+------------------------------------------------------------------------+
+| Parameter    | Possible values | Default | Description                                                            |
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: I hope it was not a huge pain in the arse to compose a table in rst ^^

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+| SWIG_JAVA    |                 | OFF     | Enables or disables libraries and native interface bindings for java   |
++--------------+-----------------+---------+------------------------------------------------------------------------+
+
+Packaging specific params
+"""""""""""""""""""""""""
+
++-----------------------+-----------------+---------+--------------------------------------------+
+| Parameter             | Possible values | Default | Description                                |
++=======================+=================+=========+============================================+
+| ENABLE_LIBS_PACKAGING |      ON/OFF     | ON      | Enables or disables all types of packaging |
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: I don't really understand why parameter ENABLE_LIBS_PACKAGING — Enables or disables all types of packaging, maybe there is something missing

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+
+Main params
+"""""""""""
+
++--------------+-----------------+---------+------------------------------------------------------------------------+
+| Parameter    | Possible values | Default | Description                                                            |
++==============+=================+=========+========================================================================+
+| TESTING      |      ON/OFF     | ON      | Enables or disables build of the tests                                 |
++--------------+                 +---------+------------------------------------------------------------------------+
+| BENCHMARKING |                 | OFF     | Enables or disables build of the benchmarks                            |
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: …which is what? 🤔 I think that we should tell that it disables build of benchmarking library, not benchmarks

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
++--------------+                 +---------+------------------------------------------------------------------------+
+| COVERAGE     |                 | OFF     | Enables or disables coverage                                           |
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: 🤔

Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: what will happen if you enable coverage

Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: @l4l

File: docs/source/getting_started/getting_started_hard_way.rst (hunk 9 above)
Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: @l4l

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
++--------------+                 +---------+------------------------------------------------------------------------+
+| SWIG_PYTHON  |                 | OFF     | Enables or disables libraries and native interface bindings for python |
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: Enables or disables build of libraries

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
++--------------+                 +---------+------------------------------------------------------------------------+
+| SWIG_JAVA    |                 | OFF     | Enables or disables libraries and native interface bindings for java   |
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: Enables or disables build of libraries

File: docs/source/getting_started/getting_started_hard_way.rst (hunk 11 above)
Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: Python

File: docs/source/getting_started/getting_started_hard_way.rst (hunk 12 above)
Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: Java

File: docs/source/getting_started/getting_started_hard_way.rst (hunk 8 above)
Comment (neewy) [reply docs/source/getting_started/getting_started_hard_way.rst]: @l4l

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
++-----------------------+                 +---------+--------------------------------------------+
+| PACKAGE_ZIP           |                 | OFF     | Enables or disables zip packaging          |
++-----------------------+                 +---------+--------------------------------------------+
+| PACKAGE_TGZ           |                 | OFF     | Enables or disables tar.gz packaging       |
++-----------------------+                 +---------+--------------------------------------------+
+| PACKAGE_RPM           |                 | OFF     | Enables or disables rpm packaging          |
++-----------------------+                 +---------+--------------------------------------------+
+| PACKAGE_DEB           |                 | OFF     | Enables or disables deb packaging          |
++-----------------------+-----------------+---------+--------------------------------------------+
+Running tests (optional)
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: Can you please put an empty line after the table

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+  cmake --build build --target test
+
+Alternatively, you can run following command in the ``build`` folder
+
+.. code-block:: shell
+
+  ctest .
+
+
+Playing around with settings
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: Initially, idea of "playing around" part was to show how changing config file can affect the behaviour of Iroha peer. It refers mostly to config file

File: docs/source/getting_started/getting_started_hard_way.rst [hunk 15]
```diff
@@ -0,0 +1,269 @@
...
+.. code:: json
+
+  {
+    "block_store_path" : "/tmp/block_store/",
+    "torii_port" : 50051,
+    "internal_port" : 50541,
+    "pg_opt" : "host=localhost port=5432 user=postgres password=mysecretpassword",
+    "redis_host" : "localhost",
+    "redis_port" : 6379,
+    "max_proposal_size" : 10,
```
Comment (neewy) [inline docs/source/getting_started/getting_started_hard_way.rst]: so, playing around means changing these values in order to find an optimal value for them. For now, you can just generally say in more details what each of it means, and how we can change them, but in the future we should expand this I think with more interesting insights

File: docs/source/getting_started/index.rst [hunk 16]
```diff
@@ -1,99 +1,123 @@
...
-Getting Iroha build
--------------------
+  docker network create iroha-network
 
+Starting PostgreSQL container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Step 1
-^^^^^^
+Now we need to run ``PostgreSQL`` in a container, attach it to the network you have created before, and exposing ports for communication:
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: expose

File: docs/source/getting_started/index.rst [hunk 17]
```diff
@@ -1,99 +1,123 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^
+We are ready to launch our Iroha container. Let's do it with the following command
 
-Step 1
-^^^^^^
+.. code-block:: shell
 
-Step 2
-^^^^^^
+  docker run -it --name iroha \
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: please explain each parameter please, especially why we are using custom entrypoint :)

File: docs/source/getting_started/index.rst [hunk 18]
```diff
@@ -1,99 +1,123 @@
...
 
-Sending transactions and queries
---------------------------------
+.. code-block:: shell
 
+  irohad --config config.docker --genesis_block genesis.block --keypair_name node0
 
-Step 1
-^^^^^^
+In usual situation, you need to provide config file, generate genesis block and keypair. However, as a part of this guide we provide an example configuration for you. Don't use it in a production though.
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: you can use attention block here

Comment (neewy) [reply docs/source/getting_started/index.rst]: https://sphinx-rtd-theme.readthedocs.io/en/latest/demo/demo.html#admonitions

File: docs/source/getting_started/index.rst [hunk 19]
```diff
@@ -1,99 +1,123 @@
...
 
-Step 2
-^^^^^^
+.. hint:: You can get more information about ``irohad`` and its launch options in this section
 
-Step 3
-^^^^^^
+Congratulations! You have an Iroha node up and running! In the next section we will test it by sending some transactions.
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: pls add 🎉

File: docs/source/getting_started/index.rst [hunk 20]
```diff
@@ -1,99 +1,123 @@
...
 
-Deploy a network of Iroha peers
--------------------------------
+Interacting with Iroha network
+------------------------------
+You can interact with Iroha using various ways. You can use our client libraries to write code in various programming languages (e.g. Java, Python, Javascript, Swift) which communicates with Iroha. Alternatively, you can use ``iroha-cli`` – our command-line tool for interacting with Iroha. As a part of this guide, let's get familiar with ``iroha-cli``
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: can you please put a note that cli that they are going to use was engineered very fast, is not having any tests, and we either need help or in the future we will deliver a better one? 😭

File: docs/source/getting_started/index.rst [hunk 21]
```diff
@@ -1,99 +1,123 @@
...
 
+Open a new terminal (note that container with Iroha and ``irohad`` should be up and running) and attach to an ``iroha`` docker container:
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: > with Iroha and ``irohad``

how this is different

File: docs/source/getting_started/index.rst [hunk 22]
```diff
@@ -1,99 +1,123 @@
...
-^^^^^^
+Now you are in the interactive shell of Iroha's container again. Now we need to launch ``iroha-cli`` and pass a account name of desired user. In our example the account ``admin``, is already created in a ``test`` domain. Let's use this account to work with Iroha.
 
+.. code-block:: shell
 
-What if bad things will happen?
--------------------------------
+  iroha-cli -account_name admin@test
 
+.. note:: Full account name has a ``@`` symbol between name and domain.
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: this is how the keypair file is also named :)

File: docs/source/getting_started/index.rst [hunk 23]
```diff
@@ -1,99 +1,123 @@
...
 
+.. code-block:: shell
 
-Step 1
-^^^^^^
+  docker volume create blockstore
 
-Step 2
-^^^^^^
+Configuring Iroha network
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: Please tell that you are actually going to run a single peer, and put a link to more advanced tutorial with several peers running in a network

File: docs/source/getting_started/index.rst [hunk 24]
```diff
@@ -1,99 +1,123 @@
...
--------------------------------
+  iroha-cli -account_name admin@test
 
+.. note:: Full account name has a ``@`` symbol between name and domain.
 
-Case 1
-^^^^^^
+Creating a first transaction
+============================
+You can see the interface of ``iroha-cli`` now. Let's create a new asset, new user and new asset in this domain, add some asset to the admin account and transfer it to other account. To achieve this, please choose option ``1. New transaction (tx)`` by writing ``tx`` or ``1`` to a console.
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: > a new asset, new user and new asset 

🥇

File: docs/source/getting_started/index.rst [hunk 25]
```diff
@@ -1,99 +1,123 @@
...
 
-Case 2
-^^^^^^
+Now you can see a list of available commands. Let's try creating a new asset. Select ``14. Create Asset (crt_ast)``. Now enter a name for your asset, for example ``coolcoin``. Next, enter a Domain ID. In our example we already have a domain ``test``, so let's use it. After it we need to enter an asset precision – amount of numbers in a fractional part.
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: so you set precision to 2 digits, but you don't tell about it

File: docs/source/getting_started/index.rst [hunk 26]
```diff
@@ -1,99 +1,123 @@
...
+
+Now we need to send our transaction to Iroha peer (``2. Send to Iroha peer (send)``). Enter peer address (in our case ``localhost``) and port (``50051``). Congratulations, your transaction is submitted and you can see your transaction hash. You can use it to check transaction's status.
+
+Go back to a terminal where ``irohad`` is running. You can see logs of your transaction.
+
+Congratulation! You have submitted your first transaction to Iroha.
+
+Creating a first query
+======================
+Now let's check if ``coolcoins`` had succesfully transfered from ``admin@test`` to ``user@test``. Go back (``b``) and choose ``2. New query (qry)``. ``7. Get Account's Assets (get_acc_ast)`` can help you to check if ``user@test`` now has some money. Form a query in a way you did with commands and ``1. Send to Iroha peer (send)``.
```
Comment (neewy) [inline docs/source/getting_started/index.rst]: transferred

Comment (neewy) [reply docs/source/getting_started/index.rst]: were succesfully transferred, it is a passive voice

Comment (neewy) [reply docs/source/getting_started/index.rst]: in a way == similarly what you did with commands

Comment (neewy) [reply docs/source/getting_started/index.rst]: Please also tell what should happen, what is the output, and cheer people up!

**Comment by laSinteZ** [comment]
Fixed everything according to review + added linebreaks + added videos 

demo at [https://html-jfhbsnwvzd.now.sh/getting_started/index.html](https://html-jfhbsnwvzd.now.sh/getting_started/index.html)

File: docs/source/getting_started/index.rst [hunk 27]
```diff
@@ -51,17 +65,29 @@ Before we run Iroha container, we should create persistent volume to store files
...
+those files, you need to clone the 
+`Iroha repository <github.com/hyperledger/iroha>`_ from Github.
 
 .. code-block:: shell
 
-  git clone -b develop https://github.com/hyperledger/iroha
+  git clone -b develop https://github.com/hyperledger/iroha --depth=1
+
+.. hint:: ``--depth-1`` option allows us to download only latest commit and
+  save some time and bandwith. If you want to get a full commit hitory, you
```
Comment (l4l) [inline docs/source/getting_started/index.rst]: hi**s**tory

File: docs/source/getting_started/index.rst [hunk 28]
```diff
@@ -75,76 +101,130 @@ We are ready to launch our Iroha container. Let's do it with the following comma
...
+- with ``$(pwd)/iroha/example:/opt/iroha_data \`` we add a folder containing
+  our prepared configuration to a docker container into ``/opt/iroha_data``.
+- ``-v blockstore:/tmp/block_store \`` adds a persistent block storage wich
+  we created before to a container, so our blocks won't be lost after we stop
+  the container
+- ``--network=iroha-network \`` adds our container to previously created
+  ``iroha-network``, so Iroha and Postgres could see each other.
+- ``--entrypoint=/bin/bash \`` Because ``hyperledger/iroha-docker`` has
+  custom script which runs after starting the container, we want to override
+  it so we can start Iroha Daemon manualy.
```
Comment (l4l) [inline docs/source/getting_started/index.rst]: manual**l**y

File: docs/source/getting_started/index.rst [hunk 29]
```diff
@@ -75,76 +101,130 @@ We are ready to launch our Iroha container. Let's do it with the following comma
...
-- with ``$(pwd)/iroha/example:/opt/iroha_data \`` we add a folder containing our prepared configuration to a docker container into ``/opt/iroha_data``.
-- ``-v blockstore:/tmp/block_store \`` adds a persistent block storage wich we created before to a container, so our blocks won't be lost after we stop the container
-- ``--network=iroha-network \`` adds our container to previously created ``iroha-network``, so Iroha and Postgres could see each other.
-- ``--entrypoint=/bin/bash \`` Because ``hyperledger/iroha-docker`` has custom script which runs after starting the container, we want to override it so we can start Iroha Daemon manualy.
-- ``hyperledger/iroha-docker:develop`` is the image which has the ``develop`` branch.
+- ``docker run -it --name iroha \`` attaches you to docker container called
+  ``iroha``
+- with ``$(pwd)/iroha/example:/opt/iroha_data \`` we add a folder containing
+  our prepared configuration to a docker container into ``/opt/iroha_data``.
+- ``-v blockstore:/tmp/block_store \`` adds a persistent block storage wich
```
Comment (l4l) [inline docs/source/getting_started/index.rst]: w**h**ich

File: docs/source/getting_started/index.rst [hunk 30]
```diff
@@ -75,76 +101,130 @@ We are ready to launch our Iroha container. Let's do it with the following comma
...
+Open a new terminal (note that Iroha container and ``irohad`` should be up and
+running) and attach to an ``iroha`` docker container:
 
 .. code-block:: shell
 
   docker exec -it iroha /bin/bash
 
-Now you are in the interactive shell of Iroha's container again. Now we need to launch ``iroha-cli`` and pass a account name of desired user. In our example the account ``admin``, is already created in a ``test`` domain. Let's use this account to work with Iroha.
+Now you are in the interactive shell of Iroha's container again. Now we need to
+launch ``iroha-cli`` and pass a account name of desired user. In our example
```
Comment (l4l) [inline docs/source/getting_started/index.rst]: a**n** account

File: docs/source/getting_started/index.rst [hunk 31]
```diff
@@ -75,76 +101,130 @@ We are ready to launch our Iroha container. Let's do it with the following comma
...
+You can see the interface of ``iroha-cli`` now. Let's create a new asset, add
+some asset to the admin account and transfer it to other account. To achieve
+this, please choose option ``1. New transaction (tx)`` by writing ``tx`` or
+``1`` to a console.
 
-Now you can see a list of available commands. Let's try creating a new asset. Select ``14. Create Asset (crt_ast)``. Now enter a name for your asset, for example ``coolcoin``. Next, enter a Domain ID. In our example we already have a domain ``test``, so let's use it. After it we need to enter an asset precision – amount of numbers in a fractional part. Let's set precision to ``2``.
+Now you can see a list of available commands. Let's try creating a new asset.
+Select ``14. Create Asset (crt_ast)``. Now enter a name for your asset, for
+example ``coolcoin``. Next, enter a Domain ID. In our example we already have a
+domain ``test``, so let's use it. After it we need to enter an asset precision
```
Comment (l4l) [inline docs/source/getting_started/index.rst]: After -> Then

File: docs/source/getting_started/index.rst [hunk 32]
```diff
@@ -153,25 +233,40 @@ Now let's check if ``coolcoins`` were successfully transferred from ``admin@test
...
 
   [2018-03-21 12:58:40.791297963][th:520][info] TransactionResponseHandler Transaction successfully sent
   Congratulation, your transaction was accepted for processing.
   Its hash is fc1c23f2de1b6fccbfe1166805e31697118b57d7bb5b1f583f2d96e78f60c241
 
-`Your transaction was accepted for processing`. Does it mean that we successfully cheated Iroha? Let's try to see transaction's status. Choose ``3. New transaction status request (st)`` and enter transaction's hash which you can got in console after previous command. Let's send it to Iroha. It replies with:
+`Your transaction was accepted for processing`. Does it mean that we
+had successfully cheated Iroha? Let's try to see transaction's status. Choose
+``3. New transaction status request (st)`` and enter transaction's hash which
+you can got in console after previous command. Let's send it to Iroha. It
```
Comment (l4l) [inline docs/source/getting_started/index.rst]: can g**e**t

**Comment by laSinteZ** [comment]
[https://html-karuzazfmu.now.sh/getting_started/index.html](https://html-karuzazfmu.now.sh/getting_started/index.html)

**Pull request merged**